### ISSUE TITLE
Data logging during events?

### ISSUE THREAD

**Issue by kevlam2706** [FRC2706/2017-2706-Robot-Code#60]
At some point (not for Toronto) it would be good to investigate how to strap an event logger to our code. Other teams do this (according to forum chatter) so it is definitely possible.

It would be nice to be able to review the system logs to determine sensor data. For example, understanding why autonomous modes did or didn't work, whether (and when) the vision system is detecting targets, what the ultrasonics and IRs are measuring, etc.

Questions I would love to have answered by going over match data logs:

- When, and how many, targets detected at the moment commands run like autonomous modes or operator assists
- Autonomous sensor measurements, especially for fancier modes like curve drive
- What the gear occupancy and peg detection IR sensors are reading at the time of gear catch/deployments
- Actual current consumption during real climbs
- Any kind of "why didn't that work?" scenario

A data logger on the driver station would be the easiest since that's where you'll want to keep logs for later review. You could have the RoboRio dumping logs directly onto its file system but that doesn't seem ideal to me.

**Comment by georgetzavelas** [comment]
Good idea @greyingjay Would this involve an application listening on a port and then logging packets sent to it?  Is [tshark](https://www.wireshark.org/docs/wsug_html_chunked/AppToolstshark.html) too generic?

**Comment by kevlam2706** [comment]
I have a suspicion that FRC/wpilib/driver station already provides some kind of framework, we just need to figure out how it works. I would be very surprised and disappointed if Smart Dashboard and Live
Window didn't come with some kind of logging.

**Comment by ryanlarkin** [comment]
In Java Smart Dashboard, there's an option to log values to a file, and in Labview Smart Dashboard, there is a way to view log files, but I don't know what's saved there

**Comment by ryanlarkin** [comment]
@georgetzavelas, I took out some of the RIOLog code, and made it print all the output, and it works on Mergius but not our practice bot (probably something that was configured when the robot was first imaged).

**Comment by georgetzavelas** [comment]
@eandr127 that's good to know, thanks!
